lib/mini_defender/detectors/public_domain_detector.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+module MiniDefender
+  module Detectors
+    class PublicDomainDetector
+      DOT = '.'
+
+      class << self
+        def public_domain?(host)
+          return false unless host
+
+          instance.valid?(host)
+        end
+
+        def instance
```

Comment:
Since this is a utility class and does not do any special job, let's instead use `static` methods and avoid the singleton pattern here

Reply:
The `public_domain_detector.rb` file is no longer necessary; as we will be using the `public_suffix` gem instead.